### ISSUE TITLE
Verify that CDI components can be added manually

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#1297]
Peter Fry says:

I been doing some Camel on EAP work for a customer and encountered some behaviour with the Camel Quartz component which was due to me using what I think is an anti-pattern with Camel CDI.

Specifically I coded:

```
@Startup
@ApplicationScoped
@ContextName("amq-cdi-context")
public class ActiveMQRouteBuilder extends RouteBuilder {

    @Override
    public void configure() throws Exception {    
        
        QuartzComponent quartzComponent = new QuartzComponent();
        quartzComponent.setPropertiesFile("/quartz.properties");
        getContext().addComponent("quartz2", quartzComponent);
```
The problem is that the component is not initialized correctly, when coded like this the method `onCamelContextStarted` isn't called. When a route is started all of the endpoints are resolved, and any required components are started BUT only if they haven't already been found in the camel context. When the component is added manually like in the code above the component is found during startup and camel assumes it has been started correctly etc.

**Comment by tdiesler** [comment]
Cannot reproduce. Please see [RouteBuilderF](https://github.com/wildfly-extras/wildfly-camel/blob/master/itests/standalone/basic/src/main/java/org/wildfly/camel/test/cdi/subA/RouteBuilderF.java)

**Comment by petenorth** [comment]
Thomas,

The project that reproduces on EAP 6.4 + Fuse 6.2.1 is here

https://github.com/petenorth/camel-cdi-quartz2-NOT-WORKING

Peter.

**Comment by jamesnetherton** [comment]
Have you tried with Fuse 6.3? camel-cdi had a major overhaul in Camel 2.17. The original implementation was know to be somewhat buggy.

**Comment by tdiesler** [comment]
Verified to work with [2.4-redhat](https://github.com/jboss-fuse/wildfly-camel/commit/68fece69c5e76e123479506b8538c679bbc9cc55)

**Comment by petenorth** [comment]
Have just verified that it works on EAP 6.4 + fuse-eap-installer-6.3.0.redhat-167.jar
so it indeed appears to be an issue resolved by the Camel 2.17 overhaul.

Thanks,

Peter.
